modules/ROOT/pages/dataweave-create-module.adoc

```diff
@@ -181,3 +181,278 @@ WeaveMod::myFunc("dataweave")
+
+[[docs-dw-lib]]
+== Writing Documentation for Custom DataWeave Modules
+
+DataWeave supports use of the AsciiDoc text format for documenting your functions, types, annotations, and the modules that contain them.
+
+=== Documenting Your DataWeave Functions
+
+DataWeave defines a multi-section template for documenting function code. The template provides sections for the following:
+
+* A description of the function.
+* Parameter descriptions in an AsciiDoc table format.
+* Example code that includes sections for any input, the DataWeave script, and the resulting output.
+
+To define parameter and example sections below the function description, the template uses AsciiDoc heading grammar:
+
+* `=== Parameters`
+* `=== Example`
+** `==== Source`
+** `==== Input`
+** `==== Output`
+
+The following function template provides guidance on documenting the function. As the template notes, some sections are optional.
+
+[source,DataWeave,linenums]
+----
+/**
+* %Replace with your function description%
+*
+*
+* %Add additional information to your function description% (optional section)
+*
+* === Parameters (optional section)
+*
+* [%header, cols="1,1,3"]
+* |===
+* | Name | Type | Description
+* | %`The parameter name`% | %`The parameter type`% | %The parameter description%  (one row per param)
+* |===
+*
+* === Example (optional section)
+*
+* %The example description% (optional)
+*
+* ==== Source (optional section)
+*
+* [source,%The language%,linenums] (optional)
+* ----
+* YOUR CODE
+* ----
+*
+* ==== Input (optional section)
+*
+* The input description (optional)
+*
+* [source,%The language%,linenums] (optional)
+* ----
+* YOUR CODE
+* ----
+*
+* ==== Output (optional section)
+*
+* %The output description% (optional)
+*
+* [source,%The language%,linenums] (optional)
+* ----
+* YOUR CODE
+* ----
+*/
+----
+
+==== Description Section
+
+This section provides a description of the function. The section starts at the top of the comments and extends to the line before the first section heading. If there is no section header, the description extends to the end of the comments for the function.
+
+A description consists of the following parts:
+
+* Short description: The first paragraph, which provides the primary description of the function.
+* Long description: Any additional information about the function. This optional description begins two line breaks below the short description.
+
+See the <<example-documentation, example>> that includes a description with sections for the parameters and examples.
+
+Each part is useful for auto-generated documentation.
+
+==== Parameters Section
+
+The optional `=== Parameters` section describes parameters of the function.
+
+The template uses Asciidoc table format to document each parameter in a table row.
+
+....
+ [%header, cols="1,13"]
+ |===
+ | Name | Description
+ | `yourParameter1` | Your description here.
+ | `yourParameter2` | Your description here.
+ |===
```

Comment:
For this section we've 2 templates:
- Legacy format, without parameter type:

[%header, cols="1,13"]
|===
| Name | Description
| `yourParameter1` | Your description here.
| `yourParameter2` | Your description here.
|===


- New format, with parameter type:
[%header, cols="1,1,3"]
|===
| Name | Type | Description
| `yourParameter1` | The parameter type | Your description here.
| `yourParameter2` | The parameter type | Your description here.
|===

IMO we should document the 2nd template or both. Because if you use the DataWeave VSCode Extension to generate your docs, you will end with the 2nd template.